src/services/admin-api.ts

```diff
@@ -72,62 +98,92 @@ export default class AdminApi {
       req.once('error', reject)
     })
 
-    let status
-    switch (req.url) {
-      case '/status':
-        status = this.getStatus()
-        break
-      case '/routing':
-        status = this.getRoutingStatus()
-        break
-      case '/accounts':
-        status = this.getAccountStatus()
-        break
-      case '/balance':
-        status = this.getBalanceStatus()
-        break
-      case '/rates':
-        status = await this.getBackendStatus()
-        break
-      case '/stats':
-        status = this.getStats()
-        break
-      default:
-        res.statusCode = 404
-        res.setHeader('Content-Type', 'text/plain')
-        res.end('Not Found')
-        return
+    const urlPrefix = (req.url || '').split('?')[0] + '$'
+    const route = this.routes.find((route) =>
+      route.method === req.method && urlPrefix.startsWith(route.match))
+    if (!route) {
+      res.statusCode = 404
+      res.setHeader('Content-Type', 'text/plain')
+      res.end('Not Found')
+      return
+    }
+
+    const resBody = await route.fn.call(this, req.url, body && JSON.parse(body))
```

Comment:
If this throws will it automatically throw a 500 or will it hang?

Reply:
`handleRequest` is called with a `catch` in `listen()` that will return a 500 on error.